### PR TITLE
feat: Clickable docs links

### LINKED ISSUE #253: Make links inside PDFs and documents clickable when viewing with Papermark

### PR DESCRIPTION
**Pull request by Aashish-Upadhyay-101**
fixes #253 

@mfts the method used to implement this is very simple, will find a good work around this later ; ) 

for now, I think it's good. 

Ready for Review

### CODE REVIEW

**Comment by mfts** [comment]
@Aashish-Upadhyay-101 wow nice work! I cannot wait to review this one 🎉

File: components/view/nav.tsx [hunk 1]
```diff
@@ -77,7 +87,29 @@ export default function Nav({
...
+                      className="text-sm flex items-start gap-2"
+                      key={index}
+                    >
+                      <span>{index + 1}.</span>
+                      <span className="underline">{link}</span>
+                    </Link>
+                  ))}
+                </DropdownMenuContent>
+              </DropdownMenu>
+            )}
```
Comment (mfts) [inline components/view/nav.tsx]:
```suggestion
          <div className="absolute inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0 space-x-4">
            {embeddedLinks && embeddedLinks.length > 0 ? (
              <DropdownMenu>
                <DropdownMenuTrigger>
                  <div className="text-sm font-semibold mr-6">
                    Embedded Links
                  </div>
                </DropdownMenuTrigger>
                <DropdownMenuContent className="space-y-2 pt-2 px-4">
                  {embeddedLinks.map((link, index) => (
                    <Link
                      href={link}
                      target="_blank"
                      className="text-sm flex items-start gap-2"
                      key={index}
                    >
                      <span>{index + 1}.</span>
                      <span className="underline">{link}</span>
                    </Link>
                  ))}
                </DropdownMenuContent>
              </DropdownMenu>
            ) : null}
```

File: components/view/nav.tsx [hunk 2]
```diff
@@ -77,7 +87,29 @@ export default function Nav({
               )}
             </div>
           </div>
-          <div className="absolute inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0 space-x-2">
+          <div className="absolute inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0 space-x-4">
+            {embeddedLinks && embeddedLinks.length > 0 && (
+              <DropdownMenu>
+                <DropdownMenuTrigger>
+                  <div className="text-sm font-semibold mr-6">
+                    Embedded Links
```
Comment (mfts) [inline components/view/nav.tsx]:
```suggestion
                    Links on Page
```

File: components/view/nav.tsx [hunk 3]
```diff
@@ -77,7 +87,29 @@ export default function Nav({
...
+                  {embeddedLinks.map((link, index) => (
+                    <Link
+                      href={link}
+                      target="_blank"
+                      className="text-sm flex items-start gap-2"
+                      key={index}
+                    >
+                      <span>{index + 1}.</span>
+                      <span className="underline">{link}</span>
+                    </Link>
```
Comment (mfts) [inline components/view/nav.tsx]: You can just use a `<DropdownMenuItem>` from shadcn/ui and the links will be styled properly. 

Please remove the numbering `<span>{index + 1}.</span>`, I don't think that's needed

File: components/view/nav.tsx [hunk 4]
```diff
@@ -77,7 +87,29 @@ export default function Nav({
...
             </div>
           </div>
-          <div className="absolute inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0 space-x-2">
+          <div className="absolute inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0 space-x-4">
+            {embeddedLinks && embeddedLinks.length > 0 && (
+              <DropdownMenu>
+                <DropdownMenuTrigger>
+                  <div className="text-sm font-semibold mr-6">
+                    Embedded Links
+                  </div>
```
Comment (mfts) [inline components/view/nav.tsx]: You can wrap this in a `<Button>` and then it will be styled well.

File: components/view/nav.tsx [hunk 5]
```diff
@@ -77,7 +87,29 @@ export default function Nav({
...
+                </DropdownMenuTrigger>
+                <DropdownMenuContent className="space-y-2 pt-2 px-4">
+                  {embeddedLinks.map((link, index) => (
+                    <Link
```
Comment (mfts) [inline components/view/nav.tsx:105]: Add `rel="noopener noreferrer"` to the `<Link>` object

File: components/view/nav.tsx [hunk 6]
```diff
@@ -5,6 +5,14 @@ import PapermarkSparkle from "../shared/icons/papermark-sparkle";
 import { Download } from "lucide-react";
 import { Brand } from "@prisma/client";
 import Image from "next/image";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuTrigger,
+} from "@/components/ui/dropdown-menu";
+import DropDown from "../web/alternatives/dropdownproto";
```
Comment (mfts) [inline components/view/nav.tsx]: this is probably not needed here

**Review by mfts — CHANGES_REQUESTED** [review]

**Comment by Aashish-Upadhyay-101** [comment]
@mfts suggested fixes done !

**Review by mfts — APPROVED** [review]
Well done! Great work 👏

giphy-2

**Pull request merged**